### PR TITLE
Add Sphinx based documentation

### LINKED ISSUE #39: Add API documentation
Hi,

I noticed that only the CLI portion of this package is documented. I my humble opinion the Python API deserves some documentation too, especially since people are looking for an alternative to pyenchant.

Mind if I give it a go? I am fairly familiar with Sphinx and readthedocs setup as well as writing documentation.

I might throw in a little Sphinx extension, while I am at it ;)

Best
-Joe

### PR DESCRIPTION
**Pull request by codingjoe**
fixes #39

### CODE REVIEW

**Comment by Findus23** [comment]
The changes look good, but when I run `python setup.py build_sphinx`, I get an empty `api_docs.html` page.
Do I need to run anything else in addition?

**Comment by codingjoe** [comment]
@Findus23 I am not done yet. The page is empty because there is no documentation. I am writing it right now. Wait for my next commit :)

File: pylanguagetool/__init__.py [hunk 1]
```diff
@@ -48,6 +49,13 @@ def init_config():
     p.add_argument("--enabled-only", action='store_true', default=False,
                    help="enable only the rules and categories whose IDs are specified with --enabled-rules or --enabled-categories"
                    )
+    p.add_argument(
+        '--pwl', '--private-word-list',
```
Comment (Findus23) [inline pylanguagetool/__init__.py]: Great idea, but I think `personal-word-list` would be a bit better name (I think it is also called that in the Languagetool browser extension).

**Comment by codingjoe** [comment]
Hey @Findus23, I though I was going a bit over board with the personal word list, but I am glad you like it.
I am currently on the run, but I'll try to finish this asap. I also need to do another sorrow review myself.

Regarding publishing the docs, I'd recommend using readthedocs.io, since this then biggest Python open source platform. Plus it builds directly from GitHub, which saves trouble.

**Comment by Findus23** [comment]
No hurry, finish it whenever you find time.

While I see that readthedocs are doing a really great job, I prefer hosting things myself and there aren't that frequent updates that setting up automation is necessarly.

File: pylanguagetool/__init__.py (hunk 1 above)
Comment (codingjoe) [reply pylanguagetool/__init__.py]: Very good point, renamed it

File: pylanguagetool/api.py [hunk 2]
```diff
@@ -1,17 +1,156 @@
...
+    Args:
+        input_text (str):
+            Plain text that will be checked for spelling mistakes.
+
+        api_url (str):
+            API base url, e.g. ``https://languagetool.org/api/v2/``
+
+        lang: Language of the given text as `RFC 3066`__ language code.
+            For example ``en-GB`` or ``de-AT``. ``auto`` is also a valid value
+            and will case the language to be detected automatically.
```
[the remark this answers is not in the document]
Comment (codingjoe) [reply pylanguagetool/api.py]: good catch!

**Comment by codingjoe** [comment]
@Findus23 I edited the last two commits to accommodate the changes we discussed. Let me know if you want me to change anything else.

**Comment by Findus23** [comment]
Thanks again for your help, I'll merge it now.

**Comment by Findus23** [comment]
And the gitlab CI also succeeded: https://git.lw1.at/open-source/pylanguagetool/-/jobs/925

**Pull request merged**